source/_posts/2015-08-17-ruby-wrappers-for-symengine.markdown

```diff
@@ -0,0 +1,235 @@
+---
+layout: post
+title: Ruby Wrappers for SymEngine, a C++ symbolic manipulation library
+date: 2015-17-08 14:30
+author: Abinash Meher
+comments: true
+categories: [SymEngine,GSOC2015,GSOC,Students]
+---
+I am sure you have heard about [Wolfram Alpha](http://www.wolframalpha.com/). Some of you would have used its [Mathematics section](http://www.wolframalpha.com/examples/Math.html) at some point of time to cross check your solution to a maths problem. If you haven't, please do. A computer algebra system (CAS) does the same thing. Algebraic computation or symbolic computation are all used interchangeably. A CAS solves problems the same way a human does, but way more quickly and precisely. There is very less chance for an error, that we humans often make.
+
+## Introduction
+My project was to write the Ruby extensions for the library [SymEngine](https://github.com/sympy/symengine) and come up with a Ruby-ish interface, after which we can use the features of SymEngine from Ruby.
+
+[SymEngine](https://github.com/sympy/symengine) is a library for symbolic computation in C++. You may ask, why SymEngine? There are other CASs that I know of. This question was indeed asked. At the beginning, the idea was to use ruby wrappers for [sage](http://www.sagemath.org/) (a mathematics software system) which uses [Pynac](http://www.sagemath.org/), an interface to [GiNaC](http://www.sagemath.org/) (another CAS). As it turns out from the benchmarks, SymEngine is much faster Pynac. What abouit directly wrapping GiNaC? SymEngine is also a bit faster than GiNac.
+
+The motivation for [SymEngine](https://github.com/sympy/symengine) itself was to develop it once and then extend it to other languages rather than doing the same thing all over again for a language that it is required in.
```

Comment:
I would say:
```md
The motivation for [SymEngine](https://github.com/sympy/symengine) itself is to develop it once in C++ and
then use it from other languages rather than doing the same thing all over again for each language that it is
required in. In particular, a long term goal is to make Sage as well as SymPy use it by default, thus
unifying the Python CAS communities. The goal of implementing the Ruby wrappers is to provide a CAS
for the Ruby community.
```